collector/default_collector.go

```diff
@@ -58,19 +56,22 @@ func (c *DefaultCollector) makeMetric(name string, value float64, valueType prom
 	return prometheus.MustNewConstMetric(desc, valueType, value, labelValues...)
 }
 
-// check that all the given paths exist and are executable files
-func CheckExecutables(paths ...string) error {
-	for _, path := range paths {
-		fileInfo, err := os.Stat(path)
-		if err != nil || os.IsNotExist(err) {
-			return errors.Errorf("'%s' does not exist", path)
-		}
-		if fileInfo.IsDir() {
-			return errors.Errorf("'%s' is a directory", path)
-		}
-		if (fileInfo.Mode() & 0111) == 0 {
-			return errors.Errorf("'%s' is not executable", path)
+// run multiple metric recording functions concurrently
```

Comment:
do we need to add concurrency? We all know that it add a layer of complexity

Reply:
the function is just 15 lines of code, it's not that complex, really, and it allows for much faster scrapes. The SAPControl web service can be veeery slow.

Reply:
ok I will review this more accurately tomorrow. A part of this looks good pr. I only a bit code :grin: